pyvrp/PenaltyManager.py

```diff
@@ -119,29 +101,53 @@ class PenaltyManager:
     ----------
     params
         PenaltyManager parameters. If not provided, a default will be used.
+    initial_penalties
+        Initial penalty values for unit load (idx 0), duration (1), and
+        distance (2) violations. Defaults to the values 20, 6, and 6. See also
+        :meth:`~init_from_data` to compute potentially better values from
+        a data instance.
     """
 
     MIN_PENALTY = 1
     MAX_PENALTY = 100_000
     FEAS_TOL = 0.05
 
-    def __init__(self, params: PenaltyParams = PenaltyParams()):
+    def __init__(
+        self,
+        params: PenaltyParams = PenaltyParams(),
+        initial_penalties: tuple[int, int, int] = (20, 6, 6),
+    ):
         self._params = params
-
-        self._penalties = np.array(
-            [
-                params.init_load_penalty,
-                params.init_time_warp_penalty,
-                params.init_dist_penalty,
-            ]
-        )
-
+        self._penalties = np.asarray(initial_penalties)
         self._feas_lists: list[list[bool]] = [
             [],  # tracks recent load feasibility
             [],  # track recent time feasibility
             [],  # track recent distance feasibility
         ]
 
+    @classmethod
+    def init_from_data(
+        cls,
+        data: ProblemData,
+        params: PenaltyParams = PenaltyParams(),
+    ) -> PenaltyManager:
```

Comment:
Using a `classmethod` for this is a bit clunky, but the added benefit is that (a) we keep a way to specify the initial penalties directly, which is nice for testing, and (b) this keeps the default initialisation of
```python
pm = PenaltyManager()
```
valid, with the same outcome. So any dependent code that did not explicitly set the initial penalty values would not be affected by the change.